### PR TITLE
refactor: Return headers in RESTController requests

### PR DESCRIPTION
**Pull request by dplewis**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-JS/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

## Issue

See discussion: https://github.com/parse-community/parse-server/pull/8766

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

File: src/__tests__/Cloud-test.js [hunk 1]
```diff
@@ -229,7 +229,7 @@ describe('CloudController', () => {
         value: 12,
         when: { __type: 'Date', iso: '2015-01-01T00:00:00.000Z' },
       },
-      { useMasterKey: true },
+      { returnStatus: true, useMasterKey: true },
```
Comment (mtrezza) [inline src/__tests__/Cloud-test.js:232]: Why shouldn't the method always return the status? It's currently also not an option.

Comment (dplewis) [reply src/__tests__/Cloud-test.js:232]: It always returns the status. This is a test to show that the option was set.

Comment (mtrezza) [reply src/__tests__/Cloud-test.js:232]: I see now, the `returnStatus` is an existing option, I thought it was a new one.

**Review by mtrezza — APPROVED** [review]
Looks good; is this ready for merge?

**Comment by dplewis** [comment]
This is ready to merge

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-alpha.2](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-alpha.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.1-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.1-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.1-alpha.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.1-alpha.1)

**Pull request merged**